apps/api/src/app/workflows-v2/usecases/extract-variables/extract-variables.usecase.ts

```diff
@@ -68,71 +71,20 @@ export class BuildPayloadSchema {
     for (const [key, value] of Object.entries(controlValue)) {
       if (isStringTipTapNode(value)) {
         processedValue[key] = this.hydrateEmailSchemaUseCase.execute({ emailEditor: value });
+      } else if (key === 'skip' && isValidRule(value as RulesLogic<AdditionalOperation>)) {
+        const fields = extractFieldsFromRules(value as RulesLogic<AdditionalOperation>)
+          .filter((field) => field.startsWith('payload.') || field.startsWith('subscriber.data.'))
+          .map((field) => `{{${field}}}`);
+
+        processedValue[key] = {
+          rules: value,
+          fields,
+        };
```

Comment:
We don't need to create custom objects with fields. As mentioned  [here](https://github.com/novuhq/novu/pull/7547/files#r1925068668), we only need to wrap the `var` property with `{{...}}`. The subsequent logic will then extract the variables using Liquid.

Reply:
yes, but we don't want to wrap the "fields" with the `{{ payload.foo }}` because it will be an invalid query

Reply:
when we extract valuables we do only that, we should not update the actual control value. 
we transform the control value to be liquid js parsed, then we extract the variables

"query" should use a clean control value and not the one used in variable extractions above.

we can do quick sync as well